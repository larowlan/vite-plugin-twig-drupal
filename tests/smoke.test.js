import Markup from "../dist/test.js"
import Error from "../dist/error.js"
import ErrorInclude from "../dist/errorInclude.js"
import drupalFunctions from "../dist/drupalFunctions.js"
import Menu from "../dist/menu.js"
import { describe, expect, it } from "vitest"

describe("Basic smoke test", () => {
  it("Should support includes", () => {
    const markup = Markup()
    expect(markup).toMatchSnapshot()
    expect(markup).toContain("Nested include")
    expect(markup).toContain("Relative include")
  })
  it("Should support variables", () => {
    const markup = Markup({ title: "Pickle Fixie" })
    expect(markup).toMatchSnapshot()
    expect(markup).toContain("Pickle Fixie")
  })
  it("Should recover from errors", () => {
    const error = Error()
    expect(error).toContain("An error occurred")
  })
  it("Should recover from include errors", () => {
    const error = ErrorInclude()
    expect(error).toContain("An error occurred")
  })
  it("Should support macros", () => {
    const markup = Menu()
    expect(markup).toContain("Contact")
    expect(markup).toMatchSnapshot()
  })
  it("Should support global context and functions", () => {
    const markup = Markup()
    expect(markup).toMatchSnapshot()
    expect(markup).toContain("Nested include")
    expect(markup).toContain("IT WORKS!")
  })
  it("Should recognise default Drupal functions", () => {
    const markup = drupalFunctions()
    expect(markup).toMatchSnapshot()
    expect(markup).toContain("Functions work")
  })
})
