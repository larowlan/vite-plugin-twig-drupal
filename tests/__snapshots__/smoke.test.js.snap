// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Basic smoke test > Should support includes 1`] = `
"<section>
  <h1>Include</h1>
  <article>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. At dignissimos fugiat inventore laborum maiores molestiae neque quia quo unde veniam?
      </article>
</section>
<section>
  <h1>Embed</h1>
  <article>
        Lorem ipsum dolor sit amet.
    <button class=\\"button--primary\\">Nested include</button>
    </article>
</section>
<section>
  <h1></h1>
  <article>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. At dignissimos fugiat inventore laborum maiores molestiae neque quia quo unde veniam?
      </article>
</section>
"
`;

exports[`Basic smoke test > Should support macros 1`] = `
"<ul>
      <li><a href=\\"/home\\">Home</a>
      </li>

      <li><a href=\\"/about\\">About</a>
          <ul>
                      <li><a href=\\"/contact\\">Contact</a>
      </li>

              </ul>
      </li>

</ul>
"
`;

exports[`Basic smoke test > Should support variables 1`] = `
"<section>
  <h1>Include</h1>
  <article>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. At dignissimos fugiat inventore laborum maiores molestiae neque quia quo unde veniam?
      </article>
</section>
<section>
  <h1>Embed</h1>
  <article>
        Lorem ipsum dolor sit amet.
    <button class=\\"button--primary\\">Nested include</button>
    </article>
</section>
<section>
  <h1>Pickle Fixie</h1>
  <article>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. At dignissimos fugiat inventore laborum maiores molestiae neque quia quo unde veniam?
      </article>
</section>
"
`;
